packages/chat-component-bindings/src/messageThreadSelector.ts

```diff
@@ -136,22 +169,47 @@ export const messageThreadSelector: MessageThreadSelector = createSelector(
             message.clientMessageId !== undefined
         )
         .filter((message) => message.content && message.content.message !== '') // TODO: deal with deleted message and remove
-        .map((message) =>
-          memoizedFn(
+        .map((message) => {
+          /** logic: Looking at message A, how do we know it's read number?
+           * Assumption: if user read the latest message, user has read all messages before that
+           * ReadReceipt behaviour: read receipt is only sent to the last message
+           * if participant read a message that is sent later than message A, then the participant has read message A
+           * how do we check if the message is sent later than message A?
+           * we compare if the readon timestamp is later than the message A sent time
+           * if last read message id is not equal to message A's id, check the read on time stamp.
+           * if the last read message is read after the message A is sent, then user should have read message A as well */
+
+          let readNumber = 0;
+          for (const k in readReceiptForEachMessage) {
```

Comment:
This part of logic is a little bit complicated, would like to see some UTs(in the future PRs) when we determine the whole logic, which will help us ensure no future bugs when someone is changing the code

Reply:
https://skype.visualstudio.com/SPOOL/_workitems/edit/2762971 
Created work item for this